dnsapi/dns_nsupdate.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+
+########  Public functions #####################
+
+#Usage: dns_nsupdate_add   _acme-challenge.www.domain.com   "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+dns_nsupdate_add() {
+  fulldomain=$1
+  txtvalue=$2
+  _checkKeyFile || return 1
+  NSUPDATE_SERVER=${NSUPDATE_SERVER:-localhost}
```

Comment:
do not use bash-only grammar :`${NSUPDATE_SERVER:-"localhost"}`
use  `if ... else ..` instead.  we must be `sh` compatible.